### PR TITLE
NON-10/NON-33: Initial DB setup

### PR DESCRIPTION
**Pull request by xoen**
- flyway configuration (added a couple of tables, we may want to move into plain classes)
- initial jpa-data configuration
- sample `ReasonType` entity/repositoty/tests to check DB wiring (we may decide to move this into a plain enum class)
- added `docker-compose-local.yml` to be able to run API locally against dev DB (and possibly other services soon)
- read DB creds from k8s Secret
- updated CircleCI config to be able to run tests against a test DB instance
- updated README with badges, etc

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,16 +14,27 @@ parameters:
...
             - gradle-{{ checksum "build.gradle.kts" }}
             - gradle-
+      - hmpps/wait_till_ready_postgres
       - run:
+          name: Run check
           command: ./gradlew check
+#          name: Run check & send results to sonarcloud
+#          command: |
+#            export GRADLE_OPTS="--add-opens=java.prefs/java.util.prefs=ALL-UNNAMED --add-opens=java.base/java.lang=ALL-UNNAMED"
+#            ./gradlew check sonar --info
```
Comment (ushkarev) [inline .circleci/config.yml:37]: we don't need this bit as we're probably not going to use sonar cloud (it hasn't proved useful enough for incentives i'd say)

File: README.md [hunk 2]
```diff
@@ -1,44 +1,32 @@
...
-If this is a HMPPS project then the project will be created as part of bootstrapping - 
-see https://github.com/ministryofjustice/dps-project-bootstrap.
+This is straight-forward as authentication is delegated down to the calling services in `dev` environment.
+Environment variables to be set are as follows:
 
-## Creating a CloudPlatform namespace
+```
+API_BASE_URL_OAUTH=https://sign-in-dev.hmpps.service.justice.gov.uk/auth
+API_BASE_URL_PRISON=https://api-dev.prison.service.justice.gov.uk
+INCENTIVES_API_CLIENT_ID=[choose a suitable hmpps-auth client]
```
Comment (Mjwillis) [inline README.md:17]: Name is wrong

File: src/main/resources/db/migration/V1_2__reason_type.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+CREATE TABLE reason_type
+(
+    code         VARCHAR(6)  NOT NULL PRIMARY KEY CHECK (length(code) >= 1),
+    description  VARCHAR(30) NOT NULL CHECK (length(description) >= 1),
+    when_created TIMESTAMPTZ NOT NULL DEFAULT now(),
+    when_updated TIMESTAMPTZ NOT NULL DEFAULT now()
+);
+
+-- https://github.com/ministryofjustice/prison-api/blob/acb4076b98c706c49edd61f21061c84eb6a837bf/src/main/resources/db/migration/data/R__2_3__REFERENCE_CODES.sql#LL5181-L5184
+INSERT INTO reason_type (code, description)
```
Comment (Mjwillis) [inline src/main/resources/db/migration/V1_2__reason_type.sql:11]: Like we discussed it might actually make sense to make them enums

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**